examples/js-sub.c

```diff
@@ -130,21 +171,48 @@ int main(int argc, char **argv)
 
     if (s == NATS_OK)
     {
-        if (pull)
+        if (pull && async)
+        {
+            jsOpts.PullSubscribeAsync.MaxMessages = (int) total;
+            jsOpts.PullSubscribeAsync.NoWait = true;
+            jsOpts.PullSubscribeAsync.Timeout = 3600 * 1000; // 1 hour
+            jsOpts.PullSubscribeAsync.FetchSize = 17;
+            jsOpts.PullSubscribeAsync.KeepAhead = 7;
```

Comment:
We agreed that the default threshold for triggering new PR is 50% of initial batch.

https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-37.md#defaults-and-constraints

Reply:
Updated to use the actual defaults in the example, with comments on what these are.